taipy/core/taipy.py

```diff
@@ -973,6 +998,104 @@ def export_scenario(
+        EntitiesToBeImportAlredyExist: If there is any entity in the import folder that has already existed.
+    """
+    entity_managers = {
+        "version": _VersionManagerFactory._build_manager,
+        "scenarios": _ScenarioManagerFactory._build_manager,
+        "jobs": _JobManagerFactory._build_manager,
+        "submission": _SubmissionManagerFactory._build_manager,
+        "cycles": _CycleManagerFactory._build_manager,
+        "sequences": _SequenceManagerFactory._build_manager,
+        "tasks": _TaskManagerFactory._build_manager,
+        "data_nodes": _DataManagerFactory._build_manager,
+    }
+
+    if isinstance(folder_path, str):
+        folder: pathlib.Path = pathlib.Path(folder_path)
+    else:
+        folder = folder_path
+
+    if not folder.exists():
+        raise FileNotFoundError(f"The import folder '{folder_path}' does not exist.")
+
+    if not (folder / "scenarios").exists():
+        raise ImportFolderDoesntContainAnyScenario(folder_path)
+
+    if not (folder / "version").exists():
+        raise ImportScenarioDoesntHaveAVersion(folder_path)
+    entity_managers["version"]()._import(next((folder / "version").iterdir()), "")
+
+    valid_entity_folders = ["version", "scenarios", "jobs", "submission", "cycles", "sequences", "tasks", "data_nodes"]
+    valid_data_folder = Config.core.storage_folder
+
+    def check_if_any_importing_entity_exists(log):
+        any_entity_exists = False
+
+        for entity_folder in valid_entity_folders:
+            if not (folder / entity_folder).exists():
+                continue
+
+            manager = entity_managers[entity_folder]()
+
+            for entity_file in (folder / entity_folder).iterdir():
+                if not entity_file.is_file():
+                    continue
+                entity_id = entity_file.stem
+                if manager._exists(entity_id):
+                    log(f"{entity_id} already exists and maybe overridden if imported.")
+                    any_entity_exists = True
+
+        return any_entity_exists
+
+    if override:
+        check_if_any_importing_entity_exists(__logger.warning)
+    else:
+        if check_if_any_importing_entity_exists(__logger.error):
+            raise EntitiesToBeImportAlredyExist(folder_path)
+
+    imported_scenario = None
+
+    for entity_folder in folder.iterdir():
+        if not entity_folder.is_dir() or entity_folder.name not in valid_entity_folders + [valid_data_folder]:
+            __logger.warning(f"{entity_folder} is not a valid Taipy folder and will not be imported.")
+            continue
+
+        # Skip the version folder as it is already checked
+        if entity_folder.name == "version":
+            continue
+
+        entity_type = entity_folder.name
+        manager = entity_managers[entity_type]()
+        for entity_file in entity_folder.iterdir():
+            imported_entity = manager._import(
+                entity_file,
```

Comment:
should we put in a try-catch to raise an error saying if a file cannot be imported?

Reply:
we now have rollback mechanism where if there is any exception, all imported entities will be deleted.